### PR TITLE
Add method and UI in test app to query granted and requestable permissions

### PR DESCRIPTION
**Pull request by felixLam**
simulator screen shot 10 06 2016 11 58 41

### CODE REVIEW

File: ISHPermissionKit/Requests/ISHPermissionRequest+All.h [hunk 1]
```diff
@@ -24,6 +27,18 @@
...
++ (ISHPermissionRequest *)requestForCategory:(ISHPermissionCategory)category;
 
+/**
+ *  Check the permission state for a set of permissions.
+ *
+ *  @param categories Set of boxed ISHPermissionCategory to check for permission state.
+ *
+ *  @return A set of boxed ISHPermissionCategory values for which the
+ *          user has granted permissions among the set of
+ *      categories given as an argument.
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequest+All.h]: Indentation

File: ISHPermissionKit/Requests/ISHPermissionRequest+All.h [hunk 2]
```diff
@@ -24,6 +27,18 @@
...
+ */
++ (NSSet<NSNumber *> *)grantedPermissionsForCategories:(NSSet<NSNumber *> *)categories;
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequest+All.h]: I think what we really wanted to know is which permissions would still trigger a prompt. We can keep this method, but we'd still need something that aligns with `setupRequestablePermissionsCategoriesFromArray:` from `ISHPermissionsViewController`. We'd probably want to move the internal implementation to this file, and call it in the permissions VC setup.

File: PermissionsTestApp/GrantedPermissionsViewController.m [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        return self.permissionsGranted;
+    }
+
+    return self.permissionsNotGranted;
+}
+
+- (NSNumber *)permissionAtIndexPath:(NSIndexPath *)indexPath {
+    NSArray *permissions = [self permissionsForSection:indexPath.section];
+
+    if (permissions.count <= indexPath.row) {
```
Comment (shagedorn) [inline PermissionsTestApp/GrantedPermissionsViewController.m]: Off by one: (count = 5), (row = 5) => Check passes > Boom!

If think `indexPath.row >= permissions.count` is easier to read.

File: PermissionsTestApp/AppDelegate.m [hunk 4]
```diff
@@ -14,7 +14,7 @@
 @import Accounts;
 
 @interface AppDelegate ()<ISHPermissionsViewControllerDataSource>
-
+@property (nonatomic) GrantedPermissionsViewController *rootViewController;
```
Comment (shagedorn) [inline PermissionsTestApp/AppDelegate.m]: Why not make this a `weak` property? Then we won't need line 51.

Comment (felixLam) [inline PermissionsTestApp/AppDelegate.m]: We would still need that to not retain self.

Comment (shagedorn) [inline PermissionsTestApp/AppDelegate.m]: ...doesn't really matter for the app delegate ;) But feel free to leave as is.

File: ISHPermissionKit/Requests/ISHPermissionRequest+All.h [hunk 5]
```diff
@@ -30,15 +30,28 @@ NS_ASSUME_NONNULL_BEGIN
...
+ *  @return An array of boxed ISHPermissionCategory values for which the
+ *          user has granted permissions among the array of
  *          categories given as an argument.
+ *          The relative sorting of the categories maintained.
  */
-+ (NSSet<NSNumber *> *)grantedPermissionsForCategories:(NSSet<NSNumber *> *)categories;
++ (NSArray<NSNumber *> *)grantedPermissionsForCategories:(NSArray<NSNumber *> *)categories;
+
+/**
+ *  Check the permission state for a set of permissions and return those that can still be requested.
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequest+All.h]: an array

File: ISHPermissionKit/Requests/ISHPermissionRequest+All.h [hunk 6]
```diff
@@ -30,15 +30,28 @@ NS_ASSUME_NONNULL_BEGIN
...
  *
- *  @param categories Set of boxed ISHPermissionCategory to check for permission state.
+ *  @param categories Array of boxed ISHPermissionCategory to check for permission state.
  *
- *  @return A set of boxed ISHPermissionCategory values for which the
- *          user has granted permissions among the set of
+ *  @return An array of boxed ISHPermissionCategory values for which the
+ *          user has granted permissions among the array of
  *          categories given as an argument.
+ *          The relative sorting of the categories maintained.
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequest+All.h]: is unchanged.

File: ISHPermissionKit/Requests/ISHPermissionRequest+All.h [hunk 7]
```diff
@@ -30,15 +30,28 @@ NS_ASSUME_NONNULL_BEGIN
...
+
+/**
+ *  Check the permission state for a set of permissions and return those that can still be requested.
+ *
+ *  @param categories Array of boxed ISHPermissionCategory to check for permission state.
+ *
+ *  @return An array of boxed ISHPermissionCategory values for which the
+ *          user can be prompted among the array of
+ *          categories given as an argument.
+ *          The relative sorting of the categories maintained.
```
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequest+All.h]: is unchanged.

**Comment by shagedorn** [comment]
👍

**Pull request merged**